extension/parquet/parquet_metadata.cpp

```diff
@@ -429,13 +433,53 @@ void ParquetMetaDataOperatorData::LoadSchemaData(ClientContext &context, const v
 	collection.InitializeScan(scan_state);
 }
 
-template <bool SCHEMA>
+void ParquetMetaDataOperatorData::BindKeyValueMetaData(vector<LogicalType> &return_types, vector<string> &names) {
+	names.emplace_back("key");
+	return_types.emplace_back(LogicalType::BLOB);
+
+	names.emplace_back("value");
+	return_types.emplace_back(LogicalType::BLOB);
+}
+
+void ParquetMetaDataOperatorData::LoadKeyValueMetaData(ClientContext &context, const vector<LogicalType> &return_types,
+                                                       const string &file_path) {
+	collection.Reset();
+	ParquetOptions parquet_options(context);
+	auto reader = make_uniq<ParquetReader>(context, file_path, parquet_options);
+	idx_t count = 0;
+	DataChunk current_chunk;
+	current_chunk.Initialize(context, return_types);
+	auto meta_data = reader->GetFileMetadata();
+
+	for (idx_t col_idx = 0; col_idx < meta_data->key_value_metadata.size(); col_idx++) {
+		auto &entry = meta_data->key_value_metadata[col_idx];
+
+		current_chunk.SetValue(0, count, Value::BLOB_RAW(entry.key));
```

Comment:
Are both the key and the value blobs or are they limited to strings?

Reply:
The standard says they are allowed to be anything, i.e. blobs, but from what I can tell, in practice its almost always utf8-encoded strings/json/text.

Reply:
If the standard says they can be anything then let's leave it as blobs ont he read-side. Perhaps on the write-side values could be either (1) cast to string, or (2) taking the literal bytes if it is of blob type? Keys always need to be utf-8 if we are using struct keys.